relations/jf/src/withdraw.rs

```diff
@@ -0,0 +1,326 @@
+use jf_primitives::{
+    circuit::merkle_tree::{Merkle3AryMembershipProofVar, RescueDigestGadget},
+    merkle_tree::{prelude::RescueSparseMerkleTree, MerkleTreeScheme, UniversalMerkleTreeScheme},
+};
+use jf_relation::{Circuit, PlonkCircuit};
+use num_bigint::BigUint;
+
+const TREE_HEIGTH: usize = 11;
+
+use crate::{
+    note::{NoteGadget, NoteType, SourcedNote},
+    shielder_types::{
+        convert_account, convert_array, Account, LeafIndex, MerkleRoot, Note, Nullifier,
+        TokenAmount, TokenId, Trapdoor,
+    },
+    CircuitField, PlonkResult, PublicInput, Relation,
+};
+
+pub struct WithdrawRelation {
+    spend_note: SourcedNote,
+    redeposit_note: SourcedNote,
+    fee: TokenAmount,
+    recipient: Account,
+    token_amount_out: TokenAmount,
+    merkle_root: MerkleRoot,
+    merkle_proof: MerkleProof,
+    leaf_index: LeafIndex,
+}
+
+impl Default for WithdrawRelation {
+    fn default() -> Self {
+        Self::new(Default::default(), Default::default())
+    }
+}
+
+#[derive(Default)]
+pub struct WithdrawPublicInput {
```

Comment:
up to you, but I'm not so sure that this is the easier way than creating directly `WithdrawRelation` (we already kinda give up this `Public/PrivateInput` in the previous PR

however, if you prefer to stay with this, we must add another constraint to the circuit, i.e. that the `token_id`s are equal in both notes - notice, that adversary doesn't have to use `WithdrawRelation` to create a malformed (valid!) relation witness

Reply:
Let me leave that as it is, and I'll try to figure out some other API during/after implementing Merge relation. Added the check.